frontend/src/components/StakeSelectorItem.vue

```diff
@@ -1,217 +1,874 @@
-    'rewardTokenName',
-    'stakeType',
-    'minimumStakeTime',
-    'estimatedYield',
-    'rewardsDuration',
-    'deprecated',
-    'rewardDistributionTimeLeft',
-    'currentRewardEarned'
-  ],
+import { isStakeType, isNftStakeType, StakeType, NftStakeType } from '../interfaces/State';
+import { toBN } from '@/utils/common';
+import { secondsToDDHHMMSS } from '../utils/date-time';
+import {Multiselect} from 'vue-multiselect';
+import {PropType} from 'vue/types/options';
+import BN from 'bignumber.js';
+import {mapActions, mapState} from 'vuex';
+import i18n from '@/i18n';
+import { TranslateResult } from 'vue-i18n';
+
+interface StoreMappedActions {
+  fetchStakeDetails(payload: {stakeType: StakeType | NftStakeType}):
+  Promise<void>;
+  stake(payload: {amount: string; stakeType: StakeType | NftStakeType}):
+  void;
+  stakeNfts(payload: {ids: string[]; stakeType: StakeType | NftStakeType}):
+  void;
+  unstake(payload: {amount: string; stakeType: StakeType | NftStakeType}):
+  void;
+  unstakeNfts(payload: {ids: string[]; stakeType: StakeType | NftStakeType}):
+  void;
+  unstakeKing(payload: {amount: string;}):
+  Promise<void>;
+  claimKingReward():
+  Promise<void>;
+  stakeUnclaimedRewards(payload: {stakeType: StakeType;}):
+  Promise<void>;
+  claimReward(payload: {stakeType: StakeType | NftStakeType;}):
+  Promise<void>;
+  getOwnedLandIdsWithTier():
+  Promise<LandIds[]>;
+  getStakedIds(payload: {stakeType: StakeType | NftStakeType;}):
+  Promise<string[]>;
+}
+
+enum CurrentState {
+  connectWallet,
+  stakeLocked,
+  contractFull,
+  amountIsTooBig,
+  waiting,
+  inputIsZero,
+  insufficientBalance,
+  notEnoughFundsInExitPool,
+  notEnoughFundsInExitPoolForFee,
+  ok,
+}
+
+enum RewardClaimState {
+  loading,
+  rewardLocked,
+  ok,
+  noReward,
+}
+
+interface LandIds {
+  id: string,
+  tier: number,
+}
+
+interface StakeData {
+  contractBalance: string;
+  currentRewardEarned: string;
+  ownBalance: string;
+  remainingCapacityForDeposit: BN;
+  remainingCapacityForWithdraw: BN;
+  rewardDistributionTimeLeft: number;
+  rewardMinimumStakeTime: number;
+  stakedBalance: string;
+  unlockTimeLeft: number;
+}
+
+type AllStakeTypes = StakeType | NftStakeType;
+
+export default Vue.extend({
+  components: {
+    Multiselect,
+  },
+  props: {
+    stakeTitle: {
+      type: String,
+    },
+    stakeType: {
+      type: String as PropType<AllStakeTypes>,
+    },
+    stakeTokenName: {
+      type: String,
+    },
+    rewardTokenName: {
+      type: String,
+    },
+    minimumStakeTime: {
+      type: Number,
+    },
+    estimatedYield: {
+      type: BN,
+    },
+    rewardsDuration: {
+      type: Number,
+    },
+    deprecated: {
+      type: Boolean,
+    },
+    rewardDistributionTimeLeft: {
+      type: Number,
+    },
+    currentRewardEarned: {
+      type: String,
+    },
+    totalStaked: {
+      type: String,
+    },
+    walletBalance: {
+      type: String,
+    },
+  },
+  data() {
+    return {
```

Comment:
It's fine, but in general we add "Data" interface and hold types there, not directly here. The interface always keeps tabs on what's missing and the IDE will inform you if the type is incorrect.

Reply:
[394edfb](https://github.com/CryptoBlades/cryptoblades/pull/1287/commits/394edfb4636310a2c04bc4448a64b27060a3d1c1)